waku/v2/protocol/waku_relay.nim

```diff
@@ -12,10 +13,26 @@ import
 logScope:
     topics = "wakurelay"
 
-const WakuRelayCodec* = "/vac/waku/relay/2.0.0-beta2"
+const
+  WakuRelayCodec* = "/vac/waku/relay/2.0.0-beta2"
+  DefaultKeepAlive = 5.minutes # 50% of the default chronosstream timeout duration
 
 type
   WakuRelay* = ref object of GossipSub
+    keepAlive*: bool
+
+proc keepAlive*(w: WakuRelay) {.async.} =
+  while w.keepAlive:
+    # Keep all mesh peers alive when idle
+    trace "Running keepalive"
+
+    for topic in w.topics.keys:
+      trace "Keepalive on topic", topic=topic
+      let mpeers = w.mesh.getOrDefault(topic)
+
+      w.broadcast(toSeq(mpeers), RPCMsg(control: some(ControlMessage(graft: @[ControlGraft(topicID: topic)]))))
```

Comment:
Any side effects of sending graft to already connected peers that we should be careful with? Re peer scoring etc

Reply:
No side-effects that I could find in specs, the code or in the local tests I've run. The `GRAFT` is simply handled by the connected peer up to [this point](https://github.com/status-im/nim-libp2p/blob/master/libp2p/protocols/pubsub/gossipsub/behavior.nim#L142) with no penalty.
That said, I think `keep-alive` should only be enabled in rare situations (i.e. `test` cluster) and be deprecated once more permanent solutions are in place (e.g. peer discovery, `ping` protocol in `libp2p`, etc.). It's also a good idea to double-check with @sinkingsugar that a `graft` from a peer already in the mesh has no negative side-effects.

Reply:
If you graft an explicit/direct peer it's a violation, why not just send a regular message instead? Or implement a ping protocol?

I would not mess with the gossip protocol itself personally. It's not the way it's meant to be used.